### PR TITLE
Doc: Fix severity effect for swagger

### LINKED ISSUE #2373: Severity.Effect should be an enum

### PR DESCRIPTION
**Pull request by kadhikari**
Ticket: https://jira.kisio.org/browse/NAVITIAII-2415

- Severity.Effect added in navitia-proto and used for disruption.severity
- StopTimeUpdateStatus used for impacted_stops.stop_time_effect
- ON_DEMAND_TRANSPORT added in SectionType
- Route.is_frequence to enum['False']

fix https://github.com/CanalTP/navitia/issues/2373
and is_frequence https://github.com/CanalTP/navitia/issues/2359

### CODE REVIEW

**Comment by kadhikari** [comment]
jenkins, test this please

File: source/type/pb_converter.cpp [hunk 1]
```diff
@@ -1015,7 +1015,35 @@ void PbCreator::Filler::fill_pb_object(const nd::Impact* impact, pbnavitia::Impa
     auto pb_severity = pb_impact->mutable_severity();
     pb_severity->set_name(impact->severity->wording);
     pb_severity->set_color(impact->severity->color);
-    pb_severity->set_effect(to_string(impact->severity->effect));
+    switch (impact->severity->effect) {
```
Comment (nonifier) [inline source/type/pb_converter.cpp]: Would it be cleaner to separate the enum conversion into a separate function ? Like :
```c++
pbnavitia::Severity_Effect get_severity_effect(nd::Effect::Effect e) {
	switch(e) 
	{
	case nd::Effect::NO_SERVICE: 
		return pbnavitia::Severity_Effect::Severity_Effect_NO_SERVICE;
    case nd::Effect::REDUCED_SERVICE: 
		return pbnavitia::Severity_Effect::Severity_Effect_REDUCED_SERVICE;
    [...]
	}
}

[...]
pb_severity->set_effect(get_severity_effect(impact->severity->effect));
```

**Review by nonifier — CHANGES_REQUESTED** [review]

**Comment by kadhikari** [comment]
Blocked by https://github.com/CanalTP/navitia-proto/pull/112

**Review by nonifier — APPROVED** [review]

**Review by mbounabi — APPROVED** [review]

File: source/jormungandr/tests/kirin_realtime_tests.py [hunk 2]
```diff
@@ -101,7 +101,7 @@ def _check_train_cancel_disruption(dis):
             is_valid_disruption(dis, chaos_disrup=False)
             assert dis['contributor'] == rt_topic
             assert dis['disruption_id'] == 'disruption_bob'
-            assert dis['severity']['effect'] == 'NO_SERVICE'
+            assert dis['severity']['effect'] == 'no_service'
```
Comment (kinnou02) [inline source/jormungandr/tests/kirin_realtime_tests.py]: You are changing the interface in a non retro compatible way, it's not possible.

**Review by kinnou02 — CHANGES_REQUESTED** [review]

**Comment by kadhikari** [comment]
jenkins test this please

**Comment by kadhikari** [comment]
jenkins test this please

File: source/jormungandr/jormungandr/interfaces/v1/fields.py [hunk 3]
```diff
@@ -221,8 +221,13 @@ def format_time(time):
         t = datetime.datetime.utcfromtimestamp(time)
         return t.strftime("%H%M%S")
 
+
 class enum_type(fields.Raw):
 
+    def __init__(self, *args, **kwargs):
+        self.lower_case = kwargs.pop('lower_case') if 'lower_case' in kwargs else True
```
Comment (xlqian) [inline source/jormungandr/jormungandr/interfaces/v1/fields.py]: kwargs.pop('lower_case', True)?

File: source/jormungandr/jormungandr/interfaces/v1/serializer/base.py [hunk 4]
```diff
@@ -111,8 +111,9 @@ class EnumField(jsonschema.Field):
     def __init__(self, pb_type=None, **kwargs):
         schema_type = kwargs.pop('schema_type') if 'schema_type' in kwargs else str
         schema_metadata = kwargs.pop('schema_metadata') if 'schema_metadata' in kwargs else {}
+        self.lower_case = kwargs.pop('lower_case') if 'lower_case' in kwargs else True
```
Comment (xlqian) [inline source/jormungandr/jormungandr/interfaces/v1/serializer/base.py]: kwargs.pop('lower_case', True)?

File: source/jormungandr/jormungandr/interfaces/v1/fields.py [hunk 5]
```diff
@@ -221,8 +221,13 @@ def format_time(time):
         t = datetime.datetime.utcfromtimestamp(time)
         return t.strftime("%H%M%S")
 
+
 class enum_type(fields.Raw):
 
+    def __init__(self, *args, **kwargs):
+        self.lower_case = kwargs.pop('lower_case', True)
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/interfaces/v1/fields.py]: why not take the parameter directly?

File: source/jormungandr/jormungandr/interfaces/v1/fields.py [hunk 6]
```diff
@@ -536,7 +544,7 @@ def format(self, value):
 
 disruption_severity = {
     "name": fields.String(),
-    "effect": fields.String(),
+    "effect": enum_type(attribute='effect', lower_case=False),
```
Comment (pbougue) [inline source/jormungandr/jormungandr/interfaces/v1/fields.py]: Why do you have to mention attribute? Not sure, but isn't that automagic?

File: source/jormungandr/jormungandr/interfaces/v1/serializer/pt.py [hunk 7]
```diff
@@ -79,7 +80,7 @@ class MessageSerializer(PbNestedSerializer):
 
 class SeveritySerializer(PbNestedSerializer):
     name = jsonschema.Field(schema_type=str)
-    effect = jsonschema.Field(schema_type=str)
+    effect = EnumField(attr='effect', pb_type=Severity.Effect, lower_case=False)
```
Comment (pbougue) [inline source/jormungandr/jormungandr/interfaces/v1/serializer/pt.py]: Same here, is `attr` necessary?

**Review by pbougue — COMMENTED** [review]
Looks good!
If you tested it's necessary, I'm OK with that PR.

File: source/jormungandr/jormungandr/interfaces/v1/fields.py (hunk 6 above)
Comment (kadhikari) [reply source/jormungandr/jormungandr/interfaces/v1/fields.py]: It's not necessary but i added that to distinct from stop_time_effect in my modifications

File: source/jormungandr/jormungandr/interfaces/v1/serializer/pt.py (hunk 7 above)
Comment (kadhikari) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/pt.py]: It's not necessary but i added that to distinct from stop_time_effect in my modifications

**Review by kinnou02 — APPROVED** [review]

**Pull request merged**